src/matcha_ml/services/analytics_service.py

```diff
@@ -51,6 +89,61 @@ def execute_analytics_event(
     return result, error_code
 
 
+def _time_event(
+    func: Callable[..., Any], *args: Any, **kwargs: Any
+) -> Tuple[Any, Exception, float, float]:
+    """Times the execution of the function decorated by track.
+
+    Args:
+        func (Callable): The function decorated by track.
+        *args (Any): arguments passed to the function.
+        **kwargs (Any): additional keyword arguments passed to the function.
+
+    Returns:
+        Tuple[Any, Exception, float, float]: The result of the call to func, the error code, the start time of execution, the end time of execution.
+    """
+    ts = perf_counter()
+    result, error_code = _execute_analytics_event(func, *args, **kwargs)
+    te = perf_counter()
+
+    return result, error_code, ts, te
+
+
+def _post_event(
+    event_name: AnalyticsEvent,
+    matcha_state_uuid: Optional[MatchaResourceProperty],
+    global_params: GlobalParameters,
+    error_code: Optional[Exception],
+    time_taken: float,
+) -> PostEvent:
+    """Posts the tracked analytics to Segment.
+
+    Args:
+        event_name (AnalyticsEvent): The enumerated name of the analytics event.
+        matcha_state_uuid (Optional[MatchaResourceProperty]): the uuid for the Matcha state file.
+        global_params (GlobalParameters): The GlobalParameters object containing the user id.
+        error_code (Optional[Exception]): The error propagated by a failing underlying command.
+        time_taken (float): The time taken for the underlying command to execute.
+
+    Returns:
+        (PostEvent): a PostEvent object representing the return of the Segment track method.
+    """
+    client = analytics.Client(WRITE_KEY, max_retries=1, debug=False)
+
+    event = client.track(
```

Comment:
Nit: `success, msg = client.track()` and then `PostEvent(success=success, message=msg)` - avoids index errors